velox/exec/benchmarks/OrderByBenchmark.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include <iostream>
+
+#include "glog/logging.h"
+#include "velox/exec/benchmarks/OrderByBenchmarkUtil.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace {
+
+struct TestCase {
+  explicit TestCase(
+      const std::string& testName,
+      size_t numRows,
+      const RowTypePtr& rowType,
+      int numKeys)
+      : testName{testName},
+        numRows{numRows},
+        rowType{rowType},
+        numKeys{numKeys} {}
+
+  std::string testName;
+  size_t numRows;
+  RowTypePtr rowType;
+  int numKeys;
+};
+
+class OrderByBenchmark {
+ public:
+  void addBenchmark(
+      const std::string& testName,
+      size_t numRows,
+      const RowTypePtr& rowType,
+      int32_t iterations,
+      int numKeys) {
+    auto testCase =
+        std::make_unique<TestCase>(testName, numRows, rowType, numKeys);
+    {
+      folly::addBenchmark(
+          __FILE__,
+          "OrderBy_" + testCase->testName,
+          [test = testCase.get(),
+           iterations = std::max(1, iterations / 10),
+           this]() {
+            auto plan = makeOrderByPlan(*test);
+            uint64_t inputNanos = 0;
+            uint64_t outputNanos = 0;
+            uint64_t finishNanos = 0;
+            auto start = getCurrentTimeMicro();
+            for (auto i = 0; i < iterations; ++i) {
+              std::shared_ptr<Task> task;
+              test::AssertQueryBuilder(plan).copyResults(pool_.get(), task);
+              auto stats = task->taskStats();
+              for (auto& pipeline : stats.pipelineStats) {
```

Comment:
Any particular reason not to use toPlanStats function from velox/exec/PlanNodeStats.h?

Reply:
PlanNodeStats combine them in one metric `cpuWallTiming`. https://github.com/facebookincubator/velox/blob/main/velox/exec/PlanNodeStats.h#L82
If you think we could collect each of them, I can help to refactor it.

Reply:
Sure, if it is helpful to keep the breakdown, let's update toPlanStats to produce both total CPU and input/output/etc. CPU.

Reply:
Added in https://github.com/facebookincubator/velox/pull/10986, thanks!